packages/components/clamped-text/clamped-text.ts

```diff
@@ -0,0 +1,210 @@
+import { KbqIcon } from '@koobiq/components/icon';
+import { KbqLinkModule } from '@koobiq/components/link';
+import { debounceTime, map, of, pairwise, skip } from 'rxjs';
+import { KBQ_CLAMPED_TEXT_LOCALE_CONFIGURATION, kbqClampedTextDefaultMaxRows } from './constants';
+
+const baseClass = 'kbq-clamped-text';
+
+@Component({
+    selector: 'kbq-clamped-text',
+    standalone: true,
+    exportAs: 'kbqClampedText',
+    imports: [
+        KbqIcon,
+        KbqButtonModule,
+        KbqLinkModule
+    ],
+    template: `
+        <div
+            #textContainer
+            class="kbq-clamped-text__content"
+            [style.-webkit-line-clamp]="lineClamp()"
+            [style.line-clamp]="lineClamp()"
+            [class.kbq-clamped-text__content_collapsed]="collapsedState()"
+        >
+            <span #text>
+                <ng-content />
+            </span>
+        </div>
+
+        @if (hasToggle()) {
+            <span
+                class="kbq-clamped-text__toggle"
+                kbq-link
+                noUnderline
+                pseudo
+                role="button"
+                (click)="toggleIsCollapsed($event)"
+                (keydown.enter)="toggleIsCollapsed($event)"
+                (keydown.space)="toggleIsCollapsed($event)"
+            >
+                @let config = localeConfiguration();
+
+                @if (collapsedState()) {
+                    <i kbq-icon="kbq-chevron-down_16"></i>
+                    <span class="kbq-link__text">{{ config!.openText }}</span>
+                } @else {
+                    <i kbq-icon="kbq-chevron-up_16"></i>
+                    <span class="kbq-link__text">{{ config!.closeText }}</span>
+                }
+            </span>
+        }
+    `,
+    styleUrls: ['./clamped-text.scss'],
+    host: {
+        class: baseClass,
+        '[attr.aria-expanded]': 'collapsedState() ? "false" : "true"'
+    },
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class KbqClampedText implements AfterViewInit {
+    /**
+     * Max rows before text is clamped.
+     * @default kbqClampedTextDefaultMaxRows
+     */
+    readonly rows = input<number>(kbqClampedTextDefaultMaxRows);
+    /** Collapsed state: `true` = collapsed, `false` = expanded, `undefined` = auto. */
+    readonly isCollapsed = input<boolean>();
+    /**
+     * Debounce time on resize observer when recalculating toggle and text visibility.
+     * @default 0
+     */
+    readonly debounceTime = input(0, { transform: numberAttribute });
+    /** Emits when collapsed state changes. Used for two-way binding with `isCollapsed`. */
+    readonly isCollapsedChange = output<boolean>();
+
+    /** @docs-private */
+    readonly text = viewChild<ElementRef<HTMLSpanElement>>('text');
+    /** @docs-private */
+    readonly textContainer = viewChild<ElementRef<HTMLDivElement>>('textContainer');
```

Comment:
необходимо указать что оба элемента являются обязательными 
```suggestion
    readonly textContainer = viewChild.required<ElementRef<HTMLDivElement>>('textContainer');
```

Reply:
в таком случае необходимо поправить `this.textContainer()!` и `this.text()!`, удалив `!`